commit e25a7f9230384a7c824888e83bcc31557aacec0a
Author: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Date:   Mon Aug 17 05:03:43 2020 -0400

    pci/linux: free the device if no kernel driver configured
    
    Patch (c79a1c67) made pci_scan_one() return without adding
    it to the list of devices on the bus.
    Since RTE_KDRV_NONE is no longer used, free the dev
    before returning.
    
    Signed-off-by: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
    Change-Id: I3bf0d4173f1dbfa7be94aacd382350b6acde907e
    Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/3817 (spdk-20.08)
    Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/5339
    Reviewed-by: Jim Harris <james.r.harris@intel.com>
    Reviewed-by: Paul Luse <paul.e.luse@intel.com>
    Reviewed-by: <dongx.yi@intel.com>
    Reviewed-by: Aleksey Marchuk <alexeymar@mellanox.com>

diff --git a/drivers/bus/pci/linux/pci.c b/drivers/bus/pci/linux/pci.c
index 2e1808b90..430938f35 100644
--- a/drivers/bus/pci/linux/pci.c
+++ b/drivers/bus/pci/linux/pci.c
@@ -332,6 +332,7 @@ pci_scan_one(const char *dirname, const struct rte_pci_addr *addr)
 			dev->kdrv = RTE_PCI_KDRV_UNKNOWN;
 	} else {
 		dev->kdrv = RTE_PCI_KDRV_NONE;
+		free(dev);
 		return 0;
 	}
 	/* device is valid, add in list (sorted) */
